### PR TITLE
Fixes and documentation to deploy the whole application in K8s

### PR DESCRIPTION
**Pull request by julienstroheker**
### WIP, Open to run CI.

TO DO before the review : 
* [X] Apply changes from #15 
* [X] Pass the C.I

About this P.R :
* [X] Remove hardcoded ips for services
* [X] Clean Dockerfiles
* [X] Add external connection string to MongoDB in dealer using Env var.
* [X] Add feed database script for MongoDB
* [X] Add PVC for the Cassandra Chart
* [X] Add API Gateway Chart
* [X] Set Always for pullpolicy
* [X] Modify prometheus Chart to make it work with the app
* [X] Add Script to deploy and run all the services on Azure/K8s

~~Feel free to review / merge if you want to.~~

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -24,6 +26,7 @@ variables:
     name: Build and Push Service Docker Image
     command: |
       TAG=$DOCKER_TAG
+      if [[ "${REGISTRY_NAMEIMAGE}" == "" ]] ; then REGISTRY_NAMEIMAGE="buildci"; fi
```
Comment (julienstroheker) [inline .circleci/config.yml]: This, will allows to build from P.R / different username whose don't have the rights from circleCI (Pull request)

File: .circleci/config.yml [hunk 2]
```diff
@@ -276,6 +278,7 @@ jobs:
           name: Build and Push Service Docker Image
           command: |
             TAG=$DOCKER_TAG
+            if [[ "${REGISTRY_NAMEIMAGE}" == "" ]] ; then REGISTRY_NAMEIMAGE="buildci"; fi
```
Comment (julienstroheker) [inline .circleci/config.yml]: Same

File: .circleci/config.yml [hunk 3]
```diff
@@ -334,6 +337,7 @@ jobs:
           command: |
             echo "Building " $GIT_FOLDER
             TAG=$DOCKER_TAG
+            if [[ "${REGISTRY_NAMEIMAGE}" == "" ]] ; then REGISTRY_NAMEIMAGE="buildci"; fi
```
Comment (julienstroheker) [inline .circleci/config.yml]: Same

**Comment by julienstroheker** [comment]
@dtzar ready to be reviewed.

I focused this PR on the `individual` helm folder. 

There is still improvement to do but at least it is a good start.

Thanks

File: .circleci/config.yml (hunk 1 above)
Comment (dtzar) [reply .circleci/config.yml]: If people fork and use CircleCI, then they will have the ability to use their own registry name via ENV variables in CircleCI.  So... this isn't needed ;)

File: deploy/helm/individual/apigateway/Chart.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+description: A Helm chart for Parts Unlimited Mrp
+name: partsunlimitedmrp
+version: 0.1.0
+sources:
+- https://dxdevop.visualstudio.com/mrpmicro/_git/code
+maintainers:
```
Comment (dtzar) [inline deploy/helm/individual/apigateway/Chart.yaml]: Please update the maintainers info to be the team :)

File: deploy/helm/individual/apigateway/Chart.yaml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+description: A Helm chart for Parts Unlimited Mrp
+name: partsunlimitedmrp
+version: 0.1.0
+sources:
```
Comment (dtzar) [inline deploy/helm/individual/apigateway/Chart.yaml:5]: Please put the github URL.

**Review by dtzar — CHANGES_REQUESTED** [review]
Overall looks good - just a few minor tweaks

File: RestAPIGateway/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+ARG VCS_REF
+ARG BUILD_DATE
+ARG port
+
+ENV expose_port=$port
+
+LABEL org.label-schema.vcs-ref=$VCS_REF \
+      org.label-schema.vendor="Microsoft" \
+      org.label-schema.name="pumrp-api" \
```
Comment (dtzar) [inline RestAPIGateway/Dockerfile:11]: For consistency, let's add org.label-schema.url and org.label-schema.vcs-url per https://github.com/Microsoft/PartsUnlimitedMRPmicro/commit/399ceaf97065b3748703c3ef0a70fcb5426aacb5

File: ZipkinServer/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+ARG VCS_REF
+ARG BUILD_DATE
+ARG port
+
+ENV expose_port=$port
+
+LABEL org.label-schema.vcs-ref=$VCS_REF \
+      org.label-schema.vendor="Microsoft" \
+      org.label-schema.name="pumrp-zipkin" \
```
Comment (dtzar) [inline ZipkinServer/Dockerfile:11]: For consistency, let's add org.label-schema.url and org.label-schema.vcs-url per https://github.com/Microsoft/PartsUnlimitedMRPmicro/commit/399ceaf97065b3748703c3ef0a70fcb5426aacb5

File: deploy/README.md [hunk 8]
```diff
@@ -0,0 +1,228 @@
+# Scripts to deploy the whole environement and the application using bash
```
Comment (dtzar) [inline deploy/README.md]: Environment spelt wrong.

File: deploy/README.md [hunk 9]
```diff
@@ -0,0 +1,228 @@
...
+
+Those steps allow you to deploy the whole environement from your local machine using Docker, AKS, ACR and MongoDB.
```
Comment (dtzar) [inline deploy/README.md]: environment.  I'd also at a minimum like to reference the automated HOL solution here in this document: https://microsoft.github.io/PartsUnlimitedMRPmicro/hols/circleci.html

File: deploy/helm/individual/apigateway/Chart.yaml (hunk 5 above)
Comment (julienstroheker) [reply deploy/helm/individual/apigateway/Chart.yaml:5]: done

File: RestAPIGateway/Dockerfile (hunk 6 above)
Comment (julienstroheker) [reply RestAPIGateway/Dockerfile:11]: done

File: ZipkinServer/Dockerfile (hunk 7 above)
Comment (julienstroheker) [reply ZipkinServer/Dockerfile:11]: done

**Comment by julienstroheker** [comment]
@dtzar review applied.

Thanks

**Review by dtzar — APPROVED** [review]

**Pull request merged**